### PR TITLE
Scenes

### PR DESCRIPTION
**Pull request by SleepingSoul**
Add scenes

### CODE REVIEW

File: SleepEngine/main.cpp [hunk 1]
```diff
@@ -11,9 +11,15 @@ int main()
 {
     EASY_PROFILER_ENABLE;
 
-	// right now the heap allocated objects leak because Updator does not own them, fixed in Updator re-do by tihran
-    auto initScene = [](slp::Game& game)
+	slp::Game game(WindowWidth, WindowHeight);
+
+    game.getClock().setFPS(60.f);
+	game.getClock().setRestrictFPS(true);
```
Comment (SleepingSoul) [inline SleepEngine/main.cpp]:
```suggestion
    game.getClock().setRestrictFPS(true);
```
Fking tabs. Have no idea where it came from

File: SleepEngine/Engine/GameTemplate/Game.cpp [hunk 2]
```diff
@@ -49,9 +98,11 @@ void Game::runFrame()
...
     m_clock.frameStart();
 
-    Base::update(m_clock.getDT());
-
-    m_renderer->render();
+    if (m_currentScene != ScenesContainer::iterator())
+    {
+        m_currentScene->second.first.update(m_clock.getDT());
+        m_renderer->render();
+    }
```
Comment (TarasMartynyuk) [inline SleepEngine/Engine/GameTemplate/Game.cpp]: it seems that default constructed iterator equals the cend() iter:
https://stackoverflow.com/questions/3395180/what-is-an-iterators-default-value
so i think it's more clear to compare it to cend(), as we always do

Comment (TarasMartynyuk) [inline SleepEngine/Engine/GameTemplate/Game.cpp]: it seems that default constructed iterator equals the cend() iter:
https://stackoverflow.com/questions/3395180/what-is-an-iterators-default-value
so i think it's more clear to compare it to cend(), as we always do

Comment (SleepingSoul) [reply SleepEngine/Engine/GameTemplate/Game.cpp]: How can it equal to cend() of an exact container if this iterator is fully independent of some exact container? I store this iterator as a member of my class and I'm sure it equals to default constructed value of this iterator)

Comment (TarasMartynyuk) [reply SleepEngine/Engine/GameTemplate/Game.cpp]: Oh, indeed its not true. But the thing you do is UB according to the second answer in my SO link

Comment (SleepingSoul) [reply SleepEngine/Engine/GameTemplate/Game.cpp]: Fixed

**Pull request merged**